File: content/problems/1620-coordinate-with-maximum-network-quality/solutions/bounded-coordinate-enumeration/solution.cpp
class Solution {
    public: vector<int> bestCoordinate(vector<vector<int>>& towers,int radius){
        int best=-1,bestX=0,bestY=0;
        for(int x=0;x<=50;++x)for(int y=0;y<=50;++y){
            int quality=0;
            for(auto& tower:towers){
                int dx=x-tower[0],dy=y-tower[1];
                double distance=sqrt(dx*dx+dy*dy);
                if(distance<=radius)quality+=(int)(tower[2]/(1+distance));
            }
            if(quality>best){
                best=quality;
                bestX=x;
                bestY=y;
            }
        }
        return {
            bestX,bestY
        };
    }
};

File: content/problems/1620-coordinate-with-maximum-network-quality/solutions/bounded-coordinate-enumeration/Solution.kt
class Solution {
    fun bestCoordinate(towers:Array<IntArray>,radius:Int):IntArray{
        var best=-1;
        var bestX=0;
        var bestY=0;
        for(x in 0..50)for(y in 0..50){
            var quality=0;
            for(tower in towers){
                val dx=x-tower[0];
                val dy=y-tower[1];
                val distance=kotlin.math.sqrt((dx*dx+dy*dy).toDouble());
                if(distance<=radius)quality+=(tower[2]/(1+distance)).toInt()
            };
            if(quality>best){
                best=quality;
                bestX=x;
                bestY=y
            }
        };
        return intArrayOf(bestX,bestY)
    }
}
